bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/handler/backend/MyBMWHttpProxy.java

```diff
@@ -0,0 +1,353 @@
+     */
+    private final String vehicleUrl;
+    private final String vehicleStateUrl;
+    private final String remoteCommandUrl;
+    private final String remoteStatusUrl;
+
+    public MyBMWHttpProxy(HttpClientFactory httpClientFactory, MyBMWBridgeConfiguration bridgeConfiguration) {
+        logger.trace("MyBMWHttpProxy - initialize");
+        httpClient = httpClientFactory.getCommonHttpClient();
+
+        myBMWTokenHandler = new MyBMWTokenController(bridgeConfiguration, httpClient);
+
+        this.bridgeConfiguration = bridgeConfiguration;
+
+        vehicleUrl = "https://" + BimmerConstants.EADRAX_SERVER_MAP.get(bridgeConfiguration.region)
+                + BimmerConstants.API_VEHICLES;
+
+        vehicleStateUrl = vehicleUrl + "/state";
+
+        remoteCommandUrl = "https://" + BimmerConstants.EADRAX_SERVER_MAP.get(bridgeConfiguration.region)
+                + BimmerConstants.API_REMOTE_SERVICE_BASE_URL;
+        remoteStatusUrl = remoteCommandUrl + "eventStatus";
+        logger.trace("MyBMWHttpProxy - ready");
+    }
+
+    @Override
+    public void setBridgeConfiguration(MyBMWBridgeConfiguration bridgeConfiguration) {
+        this.bridgeConfiguration = bridgeConfiguration;
+    }
+
+    public List<@NonNull Vehicle> requestVehicles() throws NetworkException {
+        List<@NonNull Vehicle> vehicles = new ArrayList<>();
+        List<@NonNull VehicleBase> vehiclesBase = requestVehiclesBase();
+
+        for (VehicleBase vehicleBase : vehiclesBase) {
+            VehicleStateContainer vehicleState = requestVehicleState(vehicleBase.getVin(),
+                    vehicleBase.getAttributes().getBrand());
+
+            Vehicle vehicle = new Vehicle();
+            vehicle.setVehicleBase(vehicleBase);
+            vehicle.setVehicleState(vehicleState);
+            vehicles.add(vehicle);
+        }
+
+        return vehicles;
+    }
+
+    /**
+     * request all vehicles for one specific brand and their state
+     *
+     * @param brand
+     */
+    public List<VehicleBase> requestVehiclesBase(String brand) throws NetworkException {
+        String vehicleResponseString = requestVehiclesBaseJson(brand);
+        return JsonStringDeserializer.getVehicleBaseList(vehicleResponseString);
+    }
+
+    public String requestVehiclesBaseJson(String brand) throws NetworkException {
+        byte[] vehicleResponse = get(vehicleUrl, brand, null, HTTPConstants.CONTENT_TYPE_JSON);
+        String vehicleResponseString = new String(vehicleResponse, Charset.defaultCharset());
+        return vehicleResponseString;
+    }
+
+    /**
+     * request vehicles for all possible brands
+     *
+     * @param callback
+     */
+    public List<VehicleBase> requestVehiclesBase() throws NetworkException {
+        List<VehicleBase> vehicles = new ArrayList<>();
+
+        for (String brand : BimmerConstants.REQUESTED_BRANDS) {
+            try {
+                vehicles.addAll(requestVehiclesBase(brand));
+
+                Thread.sleep(10000);
```

Comment:
Is this really needed?

Reply:
I added it as some users got the error "statusCode": 429, "message": "Rate limit is exceeded. Try again in 1 seconds."
So I added the delay to have a longer difference between two calls. It is only used during startup, so it should not be any big issue. I know it is ugly, but I had no better idea.

Reply:
OK, and it is 10 seconds on purpose? I guess a 429 could be propagated back to the handler which could then apply some logic for rescheduling. However, this would probably require quite some rework, so if it can only happen on startup, let's leave it for now.

This is only for inspiration, if you one day would like to come back to this. I don't know if this case is similar, only came to my mind from your comment about status code 429:

https://github.com/openhab/openhab-addons/blob/2be0f8dd6665ba472324ea474f85716977e3f08d/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/ApiController.java#L129-L132

Handled here:

https://github.com/openhab/openhab-addons/blob/2be0f8dd6665ba472324ea474f85716977e3f08d/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L199-L216

with exponential backoff applied here:

https://github.com/openhab/openhab-addons/blob/2be0f8dd6665ba472324ea474f85716977e3f08d/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/retry/RetryPolicyFactory.java#L42-L53

And in the end resulting in a rescheduled job here:

https://github.com/openhab/openhab-addons/blob/2be0f8dd6665ba472324ea474f85716977e3f08d/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L479-L499

Reply:
Good idea I will fix it in another PR if it is OK for you

Reply:
Yes, let's leave it as is for this PR.

Regarding 10 seconds, is that on purpose, or should it have been 1 as you mentioned "try again in 1 seconds"?

Reply:
I wanted to configure a time longer than 1s just to make sure the API should respond successfully